codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSClientConfigurationIntegration.kt

```diff
@@ -9,24 +9,36 @@ import software.amazon.smithy.aws.swift.codegen.config.AWSDefaultClientConfigura
 import software.amazon.smithy.aws.swift.codegen.config.AWSEndpointClientConfiguration
 import software.amazon.smithy.aws.swift.codegen.config.AWSRegionClientConfiguration
 import software.amazon.smithy.aws.swift.codegen.plugins.AuthSchemePlugin
+import software.amazon.smithy.aws.swift.codegen.plugins.BedrockAPIKeyPlugin
 import software.amazon.smithy.aws.swift.codegen.plugins.DefaultAWSAuthSchemePlugin
 import software.amazon.smithy.aws.swift.codegen.plugins.EndpointPlugin
+import software.amazon.smithy.aws.traits.auth.SigV4Trait
 import software.amazon.smithy.swift.codegen.config.ClientConfiguration
 import software.amazon.smithy.swift.codegen.integration.Plugin
 import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
 import software.amazon.smithy.swift.codegen.integration.ServiceConfig
 import software.amazon.smithy.swift.codegen.integration.SwiftIntegration
 import software.amazon.smithy.swift.codegen.integration.plugins.DefaultAWSClientPlugin
+import software.amazon.smithy.swift.codegen.model.getTrait
 
 class AWSClientConfigurationIntegration : SwiftIntegration {
     override fun clientConfigurations(ctx: ProtocolGenerator.GenerationContext): List<ClientConfiguration> =
         listOf(AWSDefaultClientConfiguration(), AWSRegionClientConfiguration(), AWSEndpointClientConfiguration(ctx))
 
-    override fun plugins(serviceConfig: ServiceConfig): List<Plugin> =
-        listOf(
-            DefaultAWSClientPlugin(),
-            EndpointPlugin(serviceConfig),
-            DefaultAWSAuthSchemePlugin(serviceConfig),
-            AuthSchemePlugin(serviceConfig),
-        )
+    override fun plugins(
+        ctx: ProtocolGenerator.GenerationContext,
+        serviceConfig: ServiceConfig,
+    ): List<Plugin> {
+        val list =
+            mutableListOf(
+                DefaultAWSClientPlugin(),
+                EndpointPlugin(serviceConfig),
+                DefaultAWSAuthSchemePlugin(serviceConfig),
+                AuthSchemePlugin(serviceConfig),
+            )
+        if (ctx.service.getTrait<SigV4Trait>()?.name == "bedrock") {
+            list.add(BedrockAPIKeyPlugin())
+        }
```

Comment:
The Bedrock plugin is added only for the appropriate services, i.e. ones with `bedrock` as the sigv4 signing name.